core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -89,11 +116,115 @@ public static FileSystem get(FileSystemContext context) {
       return BaseFileSystem.create(context);
     }
 
-    public static FileSystem get(ClientContext ctx) {
-      return get(FileSystemContext.create(ctx));
+    public static FileSystem create(ClientContext ctx) {
+      return create(FileSystemContext.create(ctx));
     }
   }
 
+  /**
+   * A cache for storing {@link FileSystem} clients. This should only be used by the Factory class.
+   */
+  class Cache {
+
+    private final Object mFileSystemCacheLock = new Object();
```

Comment:
Given that we are using CHM, can we get away with not using a lock?

Reply:
Yeah, we should be able to get rid of the lock.